reference/vane-apis/behn.md

```diff
@@ -0,0 +1,366 @@
++++
+title = "Behn"
+weight = 2
+template = "doc.html"
++++
+
+# Behn
+
+In this document we describe the public interface for Behn. Namely, we describe
+each `task` that Behn can be `pass`ed, and which `gift`(s) Behn can `give` in return.
+
+
+## Tasks
+
+### `%born`
+
+Each time you start your Urbit, the Arvo kernel calls the `%born` task for Behn. When
+called, Behn gets the current time from Unix and updates its list of timers
+accordingly.
+
+#### Accepts
+
+```hoon
+[~]
+```
+
+#### Returns
+
+`%born` does not return a `gift`.
+
+#### Source
+```hoon
+::  +born: urbit restarted; refresh :next-wake and store wakeup timer duct
+  ::
+  ++  born  set-unix-wake(next-wake.state ~, unix-duct.state duct)
+```
+
+
+### `%crud`
+
+`%crud` is called whenever an error involving Behn occurs, which for now means
+that Behn has no timers in its state.
+
+When called, it checks to see if the set of timers is empty. If so, it prints
+and error report containing `error`, otherwise it passes the error to `+wake`.
+
+Behn should be the first vane to be activated in a `%wake`. Occasionally this does not
+happen, but we do not yet handle this error.
+
+#### Accepts
+
+```hoon
+[tag=@tas error=tang]
+```
+Here, `tag` is a `@tas` denoting the type of error, and `error` is an error message.
+
+#### Returns
+
+If the set of timers is nonempty when Behn is `%pass`ed a `%crud` `task`, Behn
+`%give`s a `%wake` `card` to itself.
+
+```hoon
+[%wake %behn-crud-no-timer tag error]
+```
+ 
+
+#### Source
+```hoon
+++  crud
+    |=  [tag=@tas error=tang]
+    ^+  [moves state]
+    ::  behn must get activated before other vanes in a %wake
+    ::
+    ::    TODO: uncomment this case after switching %crud tags
+    ::
+    ::    We don't know how to handle other errors, so relay them to %dill
+    ::    to be printed and don't treat them as timer failures.
+    ::
+    ::  ?.  =(%wake tag)
+    ::    ~&  %behn-crud-not-first-activation^tag
+    ::    [[duct %slip %d %flog %crud tag error]~ state]
+    ::
+    ?:  =(~ timers.state)  ~|  %behn-crud-no-timer^tag^error  !!
+    ::
+    (wake `error)
+ ```
+ 
+
+### `%drip`
+
+`%drip` is utilized to delay `%give`ing a `gift`.
+
+`%drip` allows one to delay `gift`s until a given condition is met. For example,
+if Clay crashes and you do not wish for an app to then crash when it tries to
+access Clay before it is restarted, `%drip` can delay that access until Clay has
+been restored. 
```

Comment:
There is no condition.  All it does is delay giving the gift until a later event (by setting a timer for `now` and giving the gift once that timer fires).  For example, this makes it so that if an app is subscribed to Clay updates, if the app crashes while processing an update, it doesn't crash Clay (since the update came as a `%drip`, so it came in a separate event).

Reply:
I'm not sure I fully understand, how does this look? https://github.com/urbit/docs/pull/817/commits/df742bdb92325ed640e1be3d0633d59c68513003

Reply:
Looks good